packages/zoe/src/contractSupport/zoeHelpers.js

```diff
@@ -3,438 +3,363 @@
-   * entries in fromLosses are subtracted from 'from'. (If fromLosses
-   * is not defined, toGains is subtracted from 'from'.)
-   *
-   * @param {FromToAllocations} allocations - the 'to' and 'from'
-   * allocations
-   * @param {AmountKeywordRecord} toGains - what should be gained in
-   * the 'to' allocation
-   * @param {AmountKeywordRecord} [fromLosses=toGains] - what should be lost in
-   * the 'from' allocation. If not defined, fromLosses is equal to
-   * toGains. Note that the total amounts should always be equal; it
-   * is the keywords that might be different.
-   * @returns {FromToAllocations} allocations - new allocations
-   *
-   * @typedef FromToAllocations
-   * @property {AmountKeywordRecord} from
-   * @property {AmountKeywordRecord} to
-   */
-  const calcNewAllocations = (allocations, toGains, fromLosses = toGains) => {
-    const subtract = (amount, amountToSubtract) => {
+  const add = (amount, amountToAdd) => {
+    if (amount && amountToAdd) {
       const { brand } = amount;
       const amountMath = zcf.getAmountMath(brand);
-      if (amountToSubtract !== undefined) {
-        return amountMath.subtract(amount, amountToSubtract);
-      }
-      return amount;
-    };
+      return amountMath.add(amount, amountToAdd);
+    }
+    return amount || amountToAdd;
+  };
 
-    const add = (amount, amountToAdd) => {
-      if (amount && amountToAdd) {
-        const { brand } = amount;
-        const amountMath = zcf.getAmountMath(brand);
-        return amountMath.add(amount, amountToAdd);
-      }
-      return amount || amountToAdd;
-    };
+  const newFromAllocation = Object.fromEntries(
+    Object.entries(allocations.from).map(([keyword, allocAmount]) => {
+      return [keyword, subtract(allocAmount, fromLosses[keyword])];
+    }),
+  );
+
+  const allToKeywords = [
+    ...Object.keys(toGains),
+    ...Object.keys(allocations.to),
+  ];
+
+  const newToAllocation = Object.fromEntries(
+    allToKeywords.map(keyword => [
+      keyword,
+      add(allocations.to[keyword], toGains[keyword]),
+    ]),
+  );
+
+  return harden({
+    from: newFromAllocation,
+    to: newToAllocation,
+  });
+};
 
-    const newFromAllocation = Object.fromEntries(
-      Object.entries(allocations.from).map(([keyword, allocAmount]) => {
-        return [keyword, subtract(allocAmount, fromLosses[keyword])];
-      }),
-    );
-
-    const allToKeywords = [
-      ...Object.keys(toGains),
-      ...Object.keys(allocations.to),
-    ];
-
-    const newToAllocation = Object.fromEntries(
-      allToKeywords.map(keyword => [
-        keyword,
-        add(allocations.to[keyword], toGains[keyword]),
-      ]),
-    );
-
-    return harden({
-      from: newFromAllocation,
-      to: newToAllocation,
-    });
+const mergeAllocations = (currentAllocation, allocation) => {
+  const newAllocation = {
+    ...currentAllocation,
+    ...allocation,
   };
+  return newAllocation;
+};
```

Comment:
Please delete `mergeAllocations` and just open code the pattern inline at its one callsite. We use this pattern directly at plenty of other places. There's nothing about this abstraction that's specific to allocations. Reading the callsite, to know what it means, one needs to read the implementation anyway.